tests/mmm/test_delayed_saturated_mmm.py

```diff
@@ -506,3 +514,115 @@ def mock_property(self):
         ):
             DelayedSaturatedMMM.load("test_model")
         os.remove("test_model")
+
+    @pytest.mark.parametrize(
+        argnames="model_config",
+        argvalues=[
+            None,
+            {
+                "intercept": {"dist": "Normal", "kwargs": {"mu": 0, "sigma": 2}},
+                "beta_channel": {
+                    "dist": "HalfNormal",
+                    "kwargs": {"sigma": np.array([0.4533017, 0.25488063])},
+                },
+                "alpha": {
+                    "dist": "Beta",
+                    "kwargs": {
+                        "alpha": np.array([3, 3]),
+                        "beta": np.array([3.55001301, 2.87092431]),
+                    },
+                },
+                "lam": {
+                    "dist": "Gamma",
+                    "kwargs": {
+                        "alpha": np.array([3, 3]),
+                        "beta": np.array([4.12231653, 5.02896872]),
+                    },
+                },
+                "likelihood": {
+                    "dist": "StudentT",
+                    "kwargs": {"nu": 3, "sigma": 2},
+                },
+                "gamma_control": {"dist": "Normal", "kwargs": {"mu": 0, "sigma": 2}},
+                "gamma_fourier": {"dist": "Laplace", "kwargs": {"mu": 0, "b": 1}},
+            },
+        ],
+        ids=["default_config", "custom_config"],
+    )
+    def test_model_config(
+        self, model_config: Dict, toy_X: pd.DataFrame, toy_y: pd.Series
+    ):
+        # Create model instance with specified config
+        model = DelayedSaturatedMMM(
+            date_column="date",
+            channel_columns=["channel_1", "channel_2"],
+            adstock_max_lag=2,
+            yearly_seasonality=2,
+            model_config=model_config,
+        )
+
+        model.build_model(X=toy_X, y=toy_y.to_numpy())
+        # Check for default configuration
+        if model_config is None:
+            # assert observed RV type, and priors of some/all free_RVs.
+            assert isinstance(
+                model.model.observed_RVs[0].owner.op, pm.Normal
+            )  # likelihood
+            # Add more asserts as needed for default configuration
+
+        # Check for custom configuration
+        else:
+            # assert custom configuration is applied correctly
+            assert isinstance(
+                model.model.observed_RVs[0].owner.op, pm.StudentT
+            )  # likelihood
+            assert isinstance(
+                model.model.free_RVs[1].owner.op, pm.HalfNormal
```

Comment:
Small suggest, retrieve the variable by name instead of position?
```suggestion
                model.model.["var_name"].owner.op, pm.HalfNormal
```

Reply:
Yup